src/consensus/blockstore.rs

```diff
@@ -100,163 +76,60 @@ impl Blockstore {
-        trace!("reconstructed slice {slice} in slot {slot}");
-        true
     }
 
-    /// Reconstructs the block if the blockstore contains all slices.
-    ///
-    /// Returns `Some(slot, block_info)` if a block was reconstructed, `None` otherwise.
-    /// In the `Some`-case, `block_info` is the [`BlockInfo`] of the reconstructed block.
-    async fn try_reconstruct_block(&mut self, slot: Slot) -> Option<(Slot, BlockInfo)> {
-        if self.canonical_block_hash(slot).is_some() {
-            trace!("already have block for slot {slot}");
-            return None;
-        }
-        let last_slice = self.last_slices.get(&slot)?;
-        if self.stored_slices_for_slot(slot) != last_slice + 1 {
-            trace!("don't have all slices for slot {slot} yet");
-            return None;
-        }
-
-        // calculate double-Merkle tree & block hash
-        let merkle_roots: Vec<_> = self
-            .slices
-            .range(&(slot, 0)..)
-            .take_while(|(key, _)| key.0 == slot)
-            .map(|(_, s)| s.merkle_root.as_ref().unwrap())
-            .collect();
-        let tree = MerkleTree::new(&merkle_roots);
-        let block_hash = tree.get_root();
-        self.double_merkle_trees.insert(slot, tree);
-
-        // reconstruct block header
-        let first_slice = self.slices.get(&(slot, 0)).unwrap();
-        let parent_slot = u64::from_be_bytes(first_slice.data[0..8].try_into().unwrap());
-        let parent_hash = first_slice.data[8..40].try_into().unwrap();
-        self.canonical.insert(slot, block_hash);
-        // TODO: reconstruct actual block content
-        let block = Block {
-            slot,
-            block_hash,
-            parent: parent_slot,
-            parent_hash,
-            transactions: vec![],
-        };
-        let block_info = BlockInfo::from(&block);
-        self.blocks.insert((slot, block_hash), block);
-
-        // clean up raw slices
-        for slice_index in 0..=*last_slice {
-            self.slices.remove(&(slot, slice_index));
+    async fn send_votor_event(&self, event: VotorEvent) -> Option<(Slot, BlockInfo)> {
+        match event {
+            VotorEvent::FirstShred(_) => {
+                // notify Votor of first slice
+                self.votor_channel.send(event).await.unwrap();
+                None
+            }
+            VotorEvent::Block { slot, block_info } => {
+                // notify Votor of block and print block info
+                self.votor_channel.send(event).await.unwrap();
+                debug!(
+                    "reconstructed block {} in slot {} with parent {} in slot {}",
+                    &hex::encode(block_info.hash)[..8],
+                    slot,
+                    &hex::encode(block_info.parent_hash)[..8],
+                    block_info.parent_slot,
+                );
+
+                Some((slot, block_info))
+            }
+            ev => panic!("unexpected event {ev:?}"),
```

Comment:
To me this signals that in future we should consider splitting up `VotorEvent` so that we do not have to handle unexpected events here.